### PR TITLE
objective-caml (update, resolve #40676, simplify)

### LINKED ISSUE #40676: ocaml 4.02.1 build failure "[ERROR!]Arguments to this script look like '-prefix /foo/bar'"
it seems the ```gcc -march=core``` section is not quoted properly on the ```./configure``` line?
https://github.com/Homebrew/homebrew/blob/1777fc04b5afbc1ed309361c1324cfe62ce32f62/Library/Formula/objective-caml.rb#L26

```
brew install ocaml
==> Downloading http://caml.inria.fr/pub/distrib/ocaml-4.02/ocaml-4.02.1.tar.gz
Already downloaded: /home/tseemann/.cache/Homebrew/objective-caml-4.02.1.tar.gz
==> ./configure --prefix /bio/linuxbrew --mandir /bio/linuxbrew/Cellar/objective-caml/4.02.1_2/share/man -with-debug-run
-aspp
/bio/linuxbrew/bin/gcc-5 -Os -w -pipe -march=core2 -c
-no-graph

[ERROR!]Arguments to this script look like '-prefix /foo/bar', not '-prefix=/foo/bar' (note the '=').
```

### PR DESCRIPTION
**Pull request by tseemann**
Quoting issues. https://github.com/Homebrew/homebrew/issues/40676

### CODE REVIEW

File: Library/Formula/objective-caml.rb [hunk 1]
```diff
@@ -2,14 +2,17 @@ class ObjectiveCaml < Formula
...
+  sha256 "3cbc7af5a3886c8c5af8dab5568d6256a191d89ecbd4aea18eaf5b47034c6138"
   revision 2
 
   head "http://caml.inria.fr/svn/ocaml/trunk", :using => :svn
 
   option "with-x11", "Install with the Graphics module"
+
   depends_on :x11 => :optional
 
+  fails_with :clang
```
Comment (mistydemeo) [inline Library/Formula/objective-caml.rb]: Why was this added? Homebrew's been building ocaml with clang for some time.

Comment (tseemann) [inline Library/Formula/objective-caml.rb]: It's a mistake, sorry. Working on too many formula at the same time. We re-commit.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by tseemann** [comment]
Now, if only I could remember what I was *originally* trying to install before I went down this rabbit hole!

File: Library/Formula/objective-caml.rb [hunk 2]
```diff
@@ -18,20 +19,18 @@ class ObjectiveCaml < Formula
...
-      -with-debug-runtime
-    ]
-    args << "-cc" << "#{ENV.cc} #{ENV.cflags}"
-    args << "-aspp" << "#{ENV.cc} #{ENV.cflags} -c"
-    args << "-no-graph" if build.without? "x11"
-
     ENV.deparallelize # Builds are not parallel-safe, esp. with many cores
+
+    # the ./configure in this package is NOT a GNU autoconf script!
+    args = ["-prefix", prefix, "-with-debug-runtime", "-mandir", man]
```
Comment (avsm) [inline Library/Formula/objective-caml.rb]: This patch breaks OCaml quite badly, since the baked in prefix from `ocamlc -where` has changed to the Cellar location from `/usr/local/bin`.  I'll submit a pull request to change the `prefix` back to `#{HOMEBREW_PREFIX}` as it was before.

Specifically, the `camlp4` libs are no longer found (since it's a separate formula) and most packages fail to install.

Comment (mikemcquaid) [inline Library/Formula/objective-caml.rb]: @avsm If you could also add a test to the `test do` to reproduce/verify this that'd be great, thanks!

Comment (avsm) [inline Library/Formula/objective-caml.rb]: Already pushed in the next commit -- or do you mean a negative test case?

Comment (mikemcquaid) [inline Library/Formula/objective-caml.rb]: Nope, it looks good, thanks!
